test/integration/read-write-concern/write_concern.test.ts

```diff
@@ -0,0 +1,162 @@
+        expect(request.document.writeConcern.w).to.equal('majority');
+        return request.reply({ ok: 1 });
+      });
+
+      const uri = `mongodb://${server.uri()}`;
+      const client = new MongoClient(uri, { writeConcern: { w: 'majority' } });
+      return client
+        .connect()
+        .then(() => {
+          const db = client.db('wc_test');
+          const collection = db.collection('wc');
+
+          return collection.insertMany([{ a: 2 }]);
+        })
+        .then(() => {
+          return client.close();
+        });
+    });
+  });
+
+  context('when performing read operations', function () {
+    context('when writeConcern = 0', function () {
+      describe('cursor creating operations with a getMore', function () {
+        let client: MongoClient;
+        let db: Db;
+        let col: Collection;
+
+        beforeEach(async function () {
+          client = this.configuration.newClient({ writeConcern: { w: 0 } });
+          await client.connect();
+          db = client.db('writeConcernTest');
+          col = db.collection('writeConcernTest');
+
+          const docs: any[] = [];
+          for (let i = 0; i < 100; i++) {
+            docs.push({ a: i, b: i + 1 });
+          }
+
+          await col.insertMany(docs);
+        });
+
+        afterEach(async function () {
+          await db.dropDatabase();
+          await client.close();
+        });
+
+        it('succeeds on find', async function () {
+          const findResult = col.find({}, { batchSize: 2 });
+          const err = await findResult.toArray().catch(e => e);
+
+          expect(err).to.not.be.instanceOf(Error);
+        });
+
+        it('succeeds on listCollections', async function () {
+          const collections: any[] = [];
+          for (let i = 0; i < 10; i++) {
+            collections.push(`writeConcernTestCol${i + 1}`);
+          }
+
+          for (const colName of collections) {
+            await db.createCollection(colName).catch(() => null);
+          }
+
+          const cols = db.listCollections({}, { batchSize: 2 });
+
+          const err = await cols.toArray().catch(e => e);
+
+          expect(err).to.not.be.instanceOf(Error);
+        });
+
+        it('succeeds on aggregate', async function () {
+          const aggResult = col.aggregate([{ $match: { a: { $gte: 0 } } }], { batchSize: 2 });
+          const err = await aggResult.toArray().catch(e => e);
+
+          expect(err).to.not.be.instanceOf(Error);
+        });
+
+        it('succeeds on listIndexes', async function () {
+          await col.createIndex({ a: 1 });
+          await col.createIndex({ b: -1 });
+          await col.createIndex({ a: 1, b: -1 });
+
+          const listIndexesResult = col.listIndexes({ batchSize: 2 });
+          const err = await listIndexesResult.toArray().catch(e => e);
+
+          expect(err).to.not.be.instanceOf(Error);
+        });
+
+        it('succeeds on changeStream', {
```

Comment:
Ok so it's now limited to 3.6 and 4.0 of the server. Wondering why that's the case...

Reply:
Testing locally now

Reply:
Stepping through in the debugger, we're getting stuck on `src/change_stream.ts:692` at the `this.cursor.next`. Looking into what's going wrong with it.

Reply:
Maybe it's just not getting changes and version is a red herring. Maybe try refactoring it to use the events module and awaiting the next there like we do in the other tests? See: https://github.com/mongodb/node-mongodb-native/blob/main/test/integration/change-streams/change_stream.test.ts#L177